src/plugins/data_views/server/rest_api_routes/runtime_fields/create_runtime_field.ts

```diff
@@ -45,18 +45,15 @@ export const createRuntimeField = async ({
   usageCollection?.incrementCounter({ counterName });
   const dataView = await dataViewsService.get(id);
 
-  if (dataView.fields.getByName(name)) {
+  if (dataView.fields.getByName(name) || dataView.getRuntimeField(name)) {
```

Comment:
I've added an additional check but there's no way to prevent this from happening with mapped fields. ES has a 'we try to handle everything thrown at us' philosophy so its possible that `composite_runtime` exists and then there's a mapped field `composite_runtime.a`